File: packages/zodiac-dart/lib/src/western/modalities.dart
import '../models/zodiac_modality.dart';

const List<ZodiacModality> westernModalities = [
  ZodiacModality(
      id: 'cardinal',
      name: 'Cardinal',
      members: ['aries', 'cancer', 'libra', 'capricorn']),
  ZodiacModality(
      id: 'fixed',
      name: 'Fixed',
      members: ['taurus', 'leo', 'scorpio', 'aquarius']),
  ZodiacModality(
      id: 'mutable',
      name: 'Mutable',
      members: ['gemini', 'virgo', 'sagittarius', 'pisces']),
];

final Map<String, ZodiacModality> _modalityMap = {
  for (var m in westernModalities) m.id: m
};

ZodiacModality getModality(String id) {
  final mod = _modalityMap[id];
  if (mod == null) throw Exception('Modality not found: "$id"');
  return mod;
}

File: packages/zodiac-js/src/western/modalities.ts
/**
 * Western Zodiac modalities data module.
 */

import type { ZodiacModality } from '../models/zodiac-modality';

export const WESTERN_MODALITIES: readonly ZodiacModality[] = [
  { id: 'cardinal', name: 'Cardinal', members: ['aries', 'cancer', 'libra', 'capricorn'] },
  { id: 'fixed', name: 'Fixed', members: ['taurus', 'leo', 'scorpio', 'aquarius'] },
  { id: 'mutable', name: 'Mutable', members: ['gemini', 'virgo', 'sagittarius', 'pisces'] },
];

const modalityMap = new Map<string, ZodiacModality>(
  WESTERN_MODALITIES.map((m) => [m.id, m]),
);

/** Returns a modality by ID. Throws if not found. */
export function getModality(id: string): ZodiacModality {
  const mod = modalityMap.get(id);
  if (!mod) {
    throw new Error(`Modality not found: "${id}"`);
  }
  return mod;
}

/** Returns all 3 modalities. */
export function getAllModalities(): readonly ZodiacModality[] {
  return WESTERN_MODALITIES;
}
